models/solana_utils/solana_utils_daily_balances.sql

```diff
@@ -14,33 +14,35 @@
 }}
 
 WITH 
-      tokens_accounts as (
-            SELECT 
-            distinct address
-            FROM {{ source('solana','account_activity') }}
-            WHERE tx_success
-            AND token_mint_address is not null
-      )
-      
-      , updated_balances as (
+      updated_balances as (
             SELECT
-                  address 
-                  , date_trunc('day', block_time) as day
-                  , token_mint_address
+                  date_trunc('day', block_time) as day
+                  , address
+                  , coalesce(token_mint_address,original_token_mint_address) as token_mint_address
                   , cast(post_balance as double)/1e9 as sol_balance --lamport -> sol 
-                  , post_token_balance as token_balance --tokens are already correct decimals in this table
-                  , token_balance_owner
-                  , row_number() OVER (partition by address, date_trunc('day', block_time) order by block_slot desc) as latest_balance
-            FROM {{ source('solana','account_activity') }}
-            WHERE tx_success 
-            AND (
-                  --if the address is a token_mint_account, then mint_address should not be null
-                  (address NOT in (select address from tokens_accounts))
-                  or (address in (select address from tokens_accounts) AND token_mint_address is not null)
-                  )
-            {% if is_incremental() %}
-            AND block_time >= date_trunc('day', now() - interval '1' day)
-            {% endif %}
+                  , coalesce(post_token_balance,0) as token_balance --tokens are already correct decimals in this table
+                  , coalesce(token_balance_owner,original_token_balance_owner) as token_balance_owner
+                  , tx_id
+                  , block_time
+                  --could be one token mint closing and another opening on the same day. edge case but it exists.
+                  , row_number() OVER (partition by address, coalesce(token_mint_address,original_token_mint_address), date_trunc('day', block_time) order by block_slot desc) as latest_balance
```

Comment:
Can't we do `max_by` block_slot and tx id here instead and get rid of the window function? I think that should be better for performance.

Reply:
I did not know this function existed! I'm added this in now.